web/html/src/manager/virtualization/nets/properties/utils.js

```diff
@@ -0,0 +1,24 @@
+const decByte = '(?:(?:25[0-5])|(?:2[0-4][0-9])|(?:1[0-9]{2})|(?:[1-9][0-9])|(?:[0-9]))';
+
+const ipv4Regex = `(?:${decByte}\\.){3}${decByte}`;
+export const ipv4Pattern = new RegExp(`^${ipv4Regex}$`);
+
+const hexDigit = '[0-9A-Fa-f]';
+const hexWord = `${hexDigit}{1,4}`;
+const ipv6Regex = `(?:(?:${hexWord}:){7}${hexWord})|(?:(?:${hexWord}:){6}:${hexWord})|(?:(?:${hexWord}:){5}:(?:${hexWord}:)?${hexWord})|(?:(?:${hexWord}:){4}:(?:${hexWord}:){0,2}${hexWord})|(?:(?:${hexWord}:){3}:(?:${hexWord}:){0,3}${hexWord})|(?:(?:${hexWord}:){2}:(?:${hexWord}:){0,4}${hexWord})|(?:(?:${hexWord}:){6}(?:${decByte}\\.){3}${decByte})|(?:(?:${hexWord}:){0,5}:(?:${decByte}\\.){3}${decByte})|(?:::(?:${hexWord}:){0,5}(?:${decByte}\\.){3}${decByte})|(?:${hexWord}::(?:${hexWord}:){0,5}${hexWord})|(?:::(?:${hexWord}:){0,6}${hexWord})|(?:(?:${hexWord}:){1,7}:)|(?:::)`;
```

Comment:
Please add test cases for everything exported from this file, there is no way to realistically review the correctness of this and other similar regexes otherwise.

Reply:
Those are picked up from libvirt schema... I would be really surprised there is a mistake in them... and anyway I won't be able to unit test all corner cases of these, but yes, I can add a few ones.

Reply:
added! Thanks... that made me spot a typo in the UUID one ;)